### ISSUE TITLE
Consumer-side TS2595 for a dependency `.d.ts` that mixes `export =` with named exports and suppresses its own TS2309

### ISSUE THREAD

**Issue by Togetic** [microsoft/typescript-go#4773]
## Summary

When a dependency's declaration file mixes `export = X` with named exports — invalid TypeScript that
the library silences with `// @ts-ignore` — `tsc` treats the named exports as usable, while `tsgo`
additionally reports **TS2595 at the consumer's import site**. The consumer's import is correct and
cannot be fixed there, and the library's suppression cannot reach a diagnostic raised in a different
file.

Real-world impact: `consola@3.4.2` ships exactly this shape, so every
`import { consola } from "consola"` in a Nuxt project is reported as an error under `tsgo` while
`tsc` reports nothing.

## Minimal repro

`node_modules/faux/package.json`
```json
{ "name": "faux", "version": "1.0.0", "types": "./index.d.ts" }
```

`node_modules/faux/index.d.ts`
```ts
declare const thing: { info(msg: string): void };
declare function createThing(): typeof thing;
// @ts-ignore
export = thing;
export { thing, createThing };
```

`consumer.ts`
```ts
import { thing } from "faux";
thing.info("hello");
```

`tsconfig.json`
```json
{
  "compilerOptions": {
    "target": "ESNext", "module": "preserve", "moduleResolution": "Bundler",
    "strict": true, "noEmit": true, "skipLibCheck": false,
    "paths": { "faux": ["./node_modules/faux"] }
  },
  "files": ["consumer.ts"]
}
```

## Behaviour

| Declaration file | `tsc@5.9.3` | `tsgo` (`@typescript/native-preview@7.0.0-dev.20260707.2`) |
| --- | --- | --- |
| **with** the `// @ts-ignore` (as consola ships) | clean, exit 0 | `consumer.ts(1,10): error TS2595: 'thing' can only be imported by using a default import.` |
| **without** the `// @ts-ignore` | `index.d.ts(3,1): error TS2309` only | `index.d.ts(3,1): error TS2309` **and** `consumer.ts(1,10): error TS2595` |

## Expected

`tsgo` matches `tsc`: the invalid-mix error (TS2309) is raised in the declaring file, where it can be
suppressed by the library, and the named exports remain importable by consumers. No consumer-side
TS2595.

## Notes

- Not a module-resolution difference. `--traceResolution` on the same tsconfig shows `tsc` and `tsgo`
  resolving to the *identical* file. In the real-world Nuxt case the generated `paths` maps `consola`
  to a directory, so both compilers take the legacy top-level `types` field (`dist/index.d.ts`) rather
  than the `exports` map's `./dist/index.d.mts`. That is consistent between the two compilers and is
  not the cause. Verified by toggling only the `paths` entry against a package carrying an invalid
  legacy `index.d.ts` plus a valid `index.d.mts` behind an `exports` map:

  | tsconfig | `tsgo` | `tsc` |
  | --- | --- | --- |
  | with `paths` → package directory | `TS2595` at the consumer | clean |
  | without `paths` (so `exports` is consulted) | clean | clean |

  i.e. both compilers agree on *which* file to load; they disagree only on whether that file's
  suppressed `export =` conflict should also fail the consumer.
- Reproduces both with plain `tsgo` and with `vue-tsc` running on `typescript-native-bridge`
  (tsgo 7.0.2 behind the classic API), so it is not specific to one front end.
- Possibly the same family as #2781 (`import = require()` / `export =` interop, closed as *not
  planned*), but the shape here is different: the failure is a *consumer-side* diagnostic for a
  suppressed *declaration-side* error, which leaves consumers with no way to silence it.

**Comment by Togetic** [comment]
For completeness: the specific package that prompted this (`consola@3.4.2`) has been reported separately at unjs/consola#441, since its legacy `dist/index.d.ts` is genuinely invalid and fixing it there resolves the symptom independently of what is decided here.

The behaviour in this issue is the general one, though: any dependency that suppresses its own `TS2309` leaves consumers with an error they cannot suppress.

**Comment by RyanCavanaugh** [comment]
This is fully the same as #2781 - mixing named imports and export =, which is not supported.
